### ISSUE TITLE
etl::optional rvalue value_or() broken

### ISSUE THREAD

**Issue by fermentedfly** [ETLCPP/etl#720]
Hi,

I have an object whose get method returns an optional

```
struct xxx_bug_helper
{
    float value{5.0f};

    etl::optional<float> get() const
    {
        return value;
    }
};
```

If I chain `get()` and `value_or()` I get undefined behaviour

```
xxx_bug_helper helper{};
float value = helper.get().value_or(1.0); // failure, value is neither 5.0 nor 1.0
```

It works as expected if I instead assign the result of `get()` to a variable

```
xxx_bug_helper helper{};
auto temp = helper.get();
float value = temp.value_or(1.0); // value is 5.0
```

Chaining the methods worked fine previously, the new behavior may be related to changes introduced in #690 (i.e. if I delete the rvalue variant of value_or(), all is well)

Regards,
Manuel

P.s.
Sorry I can't be more helpful, but I have no idea what goes wrong here.

**Comment by jwellbelove** [comment]
It appears to be an incorrect function signature for `value_or`.
I'm fixing it in the next release, but here's the updated code.

**Comment by jwellbelove** [comment]
Dragging a zip file doesn't seem to work, so it's a txt.
[optional.txt](https://github.com/ETLCPP/etl/files/11926709/optional.txt)

**Comment by fermentedfly** [comment]
Hi,
Thanks for investigating.

**Comment by jwellbelove** [comment]
Fixed 20.37.0
